elastic/kibana
# [AO] Add data view to the new threshold rule

Closes #158840

## Summary

This PR adds selecting a persisted data view and using it in the new threshold rule.

|Flyout|Rule saved object|
|---|---|
|![image](https://github.com/elastic/kibana/assets/12370520/70bfa9d2-c4c6-44dd-bcca-ba7c83d2c433)|![image](https://github.com/elastic/kibana/assets/12370520/6f1115e2-e1f1-4348-b380-18b7ce2cacba)|

## 🧪 How to test
- Create a threshold rule with a persisted data view
    - Make sure the related feature flag is configured: `xpack.observability.unsafe.thresholdRule.enabled: true`
- Check whether the triggered alert matches the expectation related to that data view
- Check the rule saved object to ensure data is saved there correctly

## What is not covered in this PR
I will follow up on the following list in future PRs:
- [Temporary data view](https://github.com/elastic/kibana/issues/159774)
- [Initial loading](https://github.com/elastic/kibana/issues/159779)
- [Setting a timeField beside the timestamp](https://github.com/elastic/kibana/issues/159777)
- [Error handling](https://github.com/elastic/kibana/issues/159776)
- [Testing](https://github.com/elastic/kibana/issues/159778)
